README.md

```diff
@@ -182,3 +182,51 @@ as both runs will use separate caches, but providing the shared key will lead to
   with:
     prefix-key: v1
 ```
+
+## Using the error message as output
+
+In case the semver check fails, this action will populate the `error_message` output.
+
+[An output can be used in other steps](https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs), for example to comment the error message onto the pull request.
+
+```yml
+name: "Check PR"
+
+on:
+  pull_request_target:
```

Comment:
This seems a bit risky to include in an example workflow:
https://nathandavison.com/blog/github-actions-and-the-threat-of-malicious-pull-requests

Reply:
I think [sticky-pull-request-comment](https://github.com/marocchino/sticky-pull-request-comment) requires those permissions - do you think how can we suggest adding write permission in a secure way?

Also, we can remove the workflow `name` and `pull_request_target` etc. to keep the example simple (just give out `steps:`) - what do you think?

Reply:
If I remember correctly, any time a workflow has the combination of:
- `pull_request_target`
- a checkout of the PR's head repo (where you're pulling *from*)
- PR-controlled command execution (running a script, installing dependencies with `build.rs` or `setup.py` script, etc.)

Then that's a security vulnerability 100% of the time. `cargo-semver-checks` involves PR-controlled command execution because it runs `cargo doc` which installs dependencies (PR-controlled) which runs `build.rs` scripts.

It's possible I'm misunderstanding something — GitHub Actions could really have made this stuff more clear — but it'd be great to err on the side of being conservative here.

Would it be possible to wire it up so that:
- We generate the cargo-semver-checks output in a job triggered on `pull_request` not `pull_request_target`, and [save the output as an artifact](https://github.com/actions/upload-artifact).
- We then download the artifact and run the sticky-pull-request-comment job in a workflow with appropriate permissions.

This way, we don't have to give elevated permissions to the `cargo-semver-checks` run, and in the worst case an attacker might edit the comment the workflow would post on their PR instead of anything more severe.

What do you think?

Reply:
I made the suggested changes, however I didn't go with `actions/upload-artifact`, used `outputs` instead. I think the users can save the error output if they find that necessary.

This works as expected, but there is one problem:

![image](https://github.com/obi1kenobi/cargo-semver-checks-action/assets/24392180/4a17fa10-799a-40df-9173-f246b29d70fb)

Any idea how to get the correct error message?

Reply:
Okay, feels a bit hacky but I get it working (based on the example from [actions/exec](https://www.npmjs.com/package/@actions/exec#outputoptions))

One thing to mention is that `cargo-semver-checks` uses both `stderr`/`stdout` for error output so I needed to capture both. Maybe consider changing that behavior in the future to make things more consistent 🐻 i.e. I was expecting to see the full error output in `stderr` but it was not present.

Also, we can probably rename `error_message` output to something else since it captures everything now.

Another issue, `cargo-semver-checks` always uses colors, I couldn't find an argument to turn these off. (e.g. `--color=always/never` etc.)

That's why the output looks like the following:

![image](https://github.com/obi1kenobi/cargo-semver-checks-action/assets/24392180/30f59ec7-2c0b-4deb-92cf-a325e6bdf2ff)

So I stripped the ANSI codes with using another library. That was the quickest thing that I can come up with but I think we can improve it.

Right now it looks like this:

![image](https://github.com/obi1kenobi/cargo-semver-checks-action/assets/24392180/84727389-dd81-4b75-8c6b-9a2fe564f8bb)

Reply:
Thanks, this is all incredible research and top-notch feedback on what improvements to make in cargo-semver-checks in the future. Opened https://github.com/obi1kenobi/cargo-semver-checks/issues/719 for the color flags.

For the stdout/stderr changes, we'll need to both think through which things get printed where and set up test infrastructure to ensure we do it right. I'm in favor of that existing, but I'm not sure it's an immediate priority at the moment. Your approach here is very reasonable, and I'm in favor of renaming the output from `error_message` to something else — perhaps `output` or `cli_output` or something along those lines?

Reply:
03c8159386e6fcd2502199f1ce4f659b9491eae8